torch/utils/launch.py

```diff
@@ -0,0 +1,805 @@
+
+   eg: on CLX8280 with 14 instance, 4 cores per instance
+::
+
+   >>> python -m intel_extension_for_pytorch.cpu.launch  --ninstances 14 --ncore_per_instance 4 python_script args
+
+2. Run single-instance inference among multiple instances.
+   By default, runs all ninstances. If you want to independently run a single instance among ninstances, specify instance_idx.
+
+   eg: run 0th instance among SKX with 2 instance (i.e., numactl -C 0-27)
+::
+
+   >>> python -m intel_extension_for_pytorch.cpu.launch  --ninstances 2 --instance_idx 0 python_script args
+
+   eg: run 1st instance among SKX with 2 instance (i.e., numactl -C 28-55)
+::
+
+   >>> python -m intel_extension_for_pytorch.cpu.launch  --ninstances 2 --instance_idx 1 python_script args
+
+   eg: run 0th instance among SKX with 2 instance, 2 cores per instance, first four cores (i.e., numactl -C 0-1)
+::
+
+   >>> python -m intel_extension_for_pytorch.cpu.launch  --core_list "0, 1, 2, 3" --ninstances 2 --ncore_per_instance 2 --instance_idx 0 python_script args
+
+*** Distributed Training ***
+
+spawns up multiple distributed training processes on each of the training nodes. For intel_extension_for_pytorch, oneCCL
+is used as the communication backend and MPI used to launch multi-proc. To get the better
+performance, you should specify the different cores for oneCCL communication and computation
+process seperately. This tool can automatically set these ENVs(such as I_MPI_PIN_DOMIN) and launch
+multi-proc for you.
+
+The utility can be used for single-node distributed training, in which one or
+more processes per node will be spawned.  It can also be used in
+multi-node distributed training, by spawning up multiple processes on each node
+for well-improved multi-node distributed training performance as well.
+
+
+1. Single-Node multi-process distributed training
+
+::
+
+    >>> python  -m intel_extension_for_pytorch.cpu.launch --distributed  python_script  --arg1 --arg2 --arg3 and all other
+                arguments of your training script
+
+2. Multi-Node multi-process distributed training: (e.g. two nodes)
+
+
+rank 0: *(IP: 192.168.10.10, and has a free port: 295000)*
+
+::
+
+    >>> python -m intel_extension_for_pytorch.cpu.launch --distributed --nproc_per_node=xxx
+               --nnodes=2 --hostfile hostfile python_sript --arg1 --arg2 --arg3
+               and all other arguments of your training script)
+
+
+3. To look up what optional arguments this module offers:
+
+::
+
+    >>> python -m intel_extension_for_pytorch.cpu.launch --help
+
+*** Memory allocator  ***
+
+"--enable_tcmalloc" and "--enable_jemalloc" can be used to enable different memory allcator.
+
+"""
+
+class CPUinfo():
+    '''
+    Get CPU inforamation, such as cores list and NUMA information.
+    '''
+    def __init__(self):
+
+        self.cpuinfo = []
+        if platform.system() == "Windows":
+            raise RuntimeError("Windows platform is not supported!!!")
```

Comment:
Does it mean that scripts supports MacOS platform?

Reply:
Probably doesn't work on macos neither, will double check with dev team.